models/users.go

```diff
@@ -112,17 +110,17 @@ type UserParams struct {
 }
 
 // GeneratePassword creates a password for the user and assigns it to the struct
-func (p UserParams) GeneratePassword(ctx context.Context) (string, error) {
+func (p *UserParams) GeneratePassword(ctx context.Context) error {
 	tempPassword, err := password.Generate(14, 1, 1, false, false)
 	if err != nil {
-		return "", NewError(ctx, err, InternalError, PasswordGenerationErrorDescription)
+		return NewError(ctx, err, InternalError, PasswordGenerationErrorDescription)
 	}
 	p.Password = tempPassword
-	return p.Password, nil
+	return nil
 }
 
 // ValidateRegistration validates the required fields for user creation, returning validation errors for any failures
-func (p UserParams) ValidateRegistration(ctx context.Context, allowedDomains []string) []error {
+func (p *UserParams) ValidateRegistration(ctx context.Context, allowedDomains []string) []error {
```

Comment:
This line now gives the error `Struct UserParams has methods on both value and pointer receivers. Such usage is not recommended by the Go Documentation.` That's the whole problem.